imageregistry/v1/types.go

```diff
@@ -303,8 +302,65 @@ type ImageRegistryConfigStorageAzure struct {
 	// object.
 	// +optional
 	CloudName string `json:"cloudName,omitempty"`
+	// networkAccess defines the network access properties for the storage account.
+	// Defaults to type: External.
+	// +kubebuilder:default={"type": "External"}
+	// +optional
+	NetworkAccess *AzureNetworkAccess `json:"networkAccess,omitempty"`
+}
+
+type AzureNetworkAccess struct {
+	// type is the network access level to be used for the storage account.
+	// type: Internal means the storage account will be private, type: External
+	// means the storage account will be publicly accessible.
+	// When type: Internal is used, a vnetName, subNetName and privateEndpointName
+	// may optionally be specified. If unspecificed, the image registry operator
+	// will discover vnet and subnet names, and generate a privateEndpointName.
+	// Defaults to "External".
+	// +kubebuilder:default:="External"
+	Type AzureNetworkAccessType `json:"type,omitempty"`
+	// internal defines the vnet and subnet names to configure a private
+	// endpoint and connect it to the storage account in order to make it
+	// private.
+	// when type: Internal and internal is unset, the image registry operator
+	// will discover vnet and subnet names, and generate a private endpoint
+	// name.
+	// +optional
+	Internal *AzureNetworkAccessInternal `json:"internal,omitempty"`
+}
+
+type AzureNetworkAccessInternal struct {
+	// vnetName is the name of the vnet the registry operates in.
```

Comment:
Must this be the same vnet as the cluster? Is there anywhere easy that I could find this information?

Reply:
> Must this be the same vnet as the cluster?

Yes, this will usually be the same vnet as the cluster. The only important thing here is that it's the vnet the image registry pod is running in, as it is the registry that needs direct access to the storage account. So if a cluster operates within multiple vnets, users should give the vnet used by the nodes where the registry is running.

Rephrased the comment in an attempt to explain this. PTAL.

> Is there anywhere easy that I could find this information?

Yes, the machinesets for the nodes where the registry runs will contain both vnet and subnet name under `.spec.template.spec.providerSpec`. By default the same vnet and subnet names are used on all machine sets.
The registry operator provides discovery for this property however, so only advanced users should need to set this explicitly.